StreetView Composable crashes with some locations

#### Environment details

1. StreetView Composable
2. Google Pixel 6 Pro - Android 13
3. android:maps-compose:2.9.1 / com.google.maps.android:maps:3.4.0 / play-services-maps:18.1.0

#### Steps to reproduce

  1. Open StreetViewActivity in sample of this repo with this `latitude=32.429634, longitude=-96.828891`

#### Code example
The **fast path** to reproduce: Change current singapore 
```
val singapore = LatLng(1.35, 103.87)
```
to 
```
val singapore = LatLng(32.429634, -96.828891)
```

and run the sample app and navigate to Street View from Main Activity

#### Stack trace
```
com.google.maps.api.android.lib6.common.apiexception.f: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter it
	at com.google.maps.api.android.lib6.streetview.camera.i.g(:com.google.android.gms.dynamite_mapsdynamite@214218113@21.42.18 (190400-0):23)
	at com.google.maps.api.android.lib6.streetview.camera.e.run(:com.google.android.gms.dynamite_mapsdynamite@214218113@21.42.18 (190400-0):0)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7839)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter it
	at com.google.maps.android.compose.streetview.StreetViewPanoramaPropertiesNode.onAttached$lambda$3(Unknown Source:7)
	at com.google.maps.android.compose.streetview.StreetViewPanoramaPropertiesNode.$r8$lambda$u0P0-MFqw789nPtol3qohzgcP-I(Unknown Source:0)
	at com.google.maps.android.compose.streetview.StreetViewPanoramaPropertiesNode$$ExternalSyntheticLambda3.onStreetViewPanoramaChange(Unknown Source:2)
	at com.google.android.gms.maps.zzai.zzb(com.google.android.gms:play-services-maps@@18.0.0:1)
	at com.google.android.gms.maps.internal.zzbk.zza(com.google.android.gms:play-services-maps@@18.0.0:2)
	at com.google.android.gms.internal.maps.zzb.onTransact(com.google.android.gms:play-services-maps@@18.0.0:3)
	at android.os.Binder.transact(Binder.java:1064)
	at dw.aZ(:com.google.android.gms.dynamite_mapsdynamite@214218113@21.42.18 (190400-0):2)
	at com.google.maps.api.android.lib6.streetview.camera.i.g(:com.google.android.gms.dynamite_mapsdynamite@214218113@21.42.18 (190400-0):22)
	at com.google.maps.api.android.lib6.streetview.camera.e.run(:com.google.android.gms.dynamite_mapsdynamite@214218113@21.42.18 (190400-0):0) 
	at android.os.Handler.handleCallback(Handler.java:938) 
	at android.os.Handler.dispatchMessage(Handler.java:99) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7839) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
```

Following these steps will guarantee the quickest resolution possible.

Thanks!

--- a/README.md
+++ b/README.md
@@ -231,7 +231,6 @@ MarkerInfoWindow(
 ### Street View
 
 You can add a Street View given a location using the `StreetView` composable.
-To use it, provide a `StreetViewPanoramaOptions` object as follows:
 
 1. Test whether a Street View location is valid with the the
 `fetchStreetViewData` utility from the [`maps-compose-utils` library](#maps-compose-utility-library).
@@ -241,7 +240,7 @@ To use it, provide a `StreetViewPanoramaOptions` object as follows:
     fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
 ```
 
-2. Once the location is confirmed valid, add a Street View composable.
+2. Once the location is confirmed valid, add a Street View composable by providing a `StreetViewPanoramaOptions` object.
 
 ```kotlin
 val singapore = LatLng(1.3588227, 103.8742114)

--- a/README.md
+++ b/README.md
@@ -241,6 +241,17 @@ StreetView(
 )
 ```
 
+In order to test whether that Street View location is valid, we recommend using the
+`fetchStreetViewData` utility from the Android Map Utils library. We showcase an example of this 
+in `StreetViewActivity`:
+
+```kotlin
+ streetViewResult =
+    fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
+```
+
+Please make sure that this API Key has Street View Static API activated.
+
 </details>
 
 <details>

--- a/README.md
+++ b/README.md
@@ -24,7 +24,8 @@ You no longer need to specify the Maps SDK for Android or its Utility Library as
 dependencies {
     implementation 'com.google.maps.android:maps-compose:3.1.1'
 
-    // Optionally, you can include the Compose utils library for Clustering, etc.
+    // Optionally, you can include the Compose utils library for Clustering,
+    // Street View metadata checks, etc.
     implementation 'com.google.maps.android:maps-compose-utils:3.1.1'
 
     // Optionally, you can include the widgets library for ScaleBar, etc.
@@ -232,6 +233,16 @@ MarkerInfoWindow(
 You can add a Street View given a location using the `StreetView` composable.
 To use it, provide a `StreetViewPanoramaOptions` object as follows:
 
+1. Test whether a Street View location is valid with the the
+`fetchStreetViewData` utility from the [`maps-compose-utils` library](#maps-compose-utility-library).
+
+```kotlin
+ streetViewResult =
+    fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
+```
+
+2. Once the location is confirmed valid, add a Street View composable.
+
 ```kotlin
 val singapore = LatLng(1.3588227, 103.8742114)
 StreetView(
@@ -240,17 +251,6 @@ StreetView(
     }
 )
 ```
-
-In order to test whether that Street View location is valid, we recommend using the
-`fetchStreetViewData` utility from the [Maps SDK for Android Utility Library](https://github.com/googlemaps/android-maps-utils). We showcase an example of this
-in [`StreetViewActivity`](https://github.com/googlemaps/android-maps-compose/blob/main/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt) of the demo app:
-
-```kotlin
- streetViewResult =
-    fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
-```
-
-Be sure to [enable Street View Static API](https://goo.gle/enable-sv-static-api) on the project associated with your API key.
 
 </details>
 
@@ -275,9 +275,9 @@ GoogleMap(
 
 </details>
 
-## Utility Library
+## Maps Compose Utility Library
 
-This library also provides optional utilities in the `maps-compose-utils` library from the [Maps SDK for Android Utility Library](https://github.com/googlemaps/android-maps-utils).
+This library provides optional utilities in the `maps-compose-utils` library from the [Maps SDK for Android Utility Library](https://github.com/googlemaps/android-maps-utils).
 
 ### Clustering
 
@@ -300,7 +300,22 @@ Clustering(
 )
 ```
 
-## Widgets
+### Street View metadata utility
+
+The `fetchStreetViewData` method provides functionality to check whether a location is supported in StreetView. You can avoid errors when adding a Street View panorama to an Android app by calling this metadata utility and only adding a Street View panorama if the response is OK.
+
+> [!IMPORTANT]
+> Be sure to [enable Street View Static API](https://goo.gle/enable-sv-static-api) on the project associated with your API key.
+
+You can see example usage
+in the [`StreetViewActivity`](https://github.com/googlemaps/android-maps-compose/blob/main/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt) of the demo app:
+
+```kotlin
+ streetViewResult =
+    fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
+```
+
+## Maps Compose Widgets
 
 This library also provides optional composable widgets in the `maps-compose-widgets` library that you can use alongside the `GoogleMap` composable.
 

--- a/README.md
+++ b/README.md
@@ -23,7 +23,7 @@ You no longer need to specify the Maps SDK for Android or its Utility Library as
 ```groovy
 dependencies {
     implementation 'com.google.maps.android:maps-compose:3.1.1'
-    
+
     // Optionally, you can include the Compose utils library for Clustering, etc.
     implementation 'com.google.maps.android:maps-compose-utils:3.1.1'
 
@@ -242,15 +242,15 @@ StreetView(
 ```
 
 In order to test whether that Street View location is valid, we recommend using the
-`fetchStreetViewData` utility from the Android Map Utils library. We showcase an example of this 
-in `StreetViewActivity`:
+`fetchStreetViewData` utility from the [Maps SDK for Android Utility Library](https://github.com/googlemaps/android-maps-utils). We showcase an example of this
+in [`StreetViewActivity`](https://github.com/googlemaps/android-maps-compose/blob/main/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt) of the demo app:
 
 ```kotlin
  streetViewResult =
     fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
 ```
 
-Please make sure that this API Key has Street View Static API activated.
+Be sure to [enable Street View Static API](https://goo.gle/enable-sv-static-api) on the project associated with your API key.
 
 </details>
 
@@ -277,7 +277,7 @@ GoogleMap(
 
 ## Utility Library
 
-This library also provides optional utilities in the `maps-compose-utils` library.
+This library also provides optional utilities in the `maps-compose-utils` library from the [Maps SDK for Android Utility Library](https://github.com/googlemaps/android-maps-utils).
 
 ### Clustering
 
@@ -315,8 +315,8 @@ The [ScaleBarActivity](app/src/main/java/com/google/maps/android/compose/ScaleBa
 Both versions of this widget leverage the `CameraPositionState` in `maps-compose` and therefore are very simple to configure with their defaults:
 
 ```kotlin
-Box(Modifier.fillMaxSize()) { 
-    
+Box(Modifier.fillMaxSize()) {
+
     GoogleMap(
         modifier = Modifier.fillMaxSize(),
         cameraPositionState = cameraPositionState
@@ -339,7 +339,7 @@ Box(Modifier.fillMaxSize()) {
             .align(Alignment.TopStart),
         cameraPositionState = cameraPositionState
     )
-} 
+}
 ```
 
 The colors of the text, line, and shadow are also all configurable (e.g., based on `isSystemInDarkTheme()` on a dark map). Similarly, the `DisappearingScaleBar` animations can be configured.

--- a/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
+++ b/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
@@ -60,7 +60,6 @@ class StreetViewActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-
         setContent {
             var isPanningEnabled by remember { mutableStateOf(false) }
             var isZoomEnabled by remember { mutableStateOf(false) }
@@ -103,19 +102,21 @@ class StreetViewActivity : ComponentActivity() {
                             Log.d(TAG, "Street view long clicked")
                         }
                     )
-                }
-                Column(
-                    Modifier
-                        .fillMaxWidth()
-                        .background(Color.White)
-                        .padding(8.dp)
-                ) {
-                    StreetViewSwitch(title = "Panning", checked = isPanningEnabled) {
-                        isPanningEnabled = it
+                    Column(
+                        Modifier
+                            .fillMaxWidth()
+                            .background(Color.White)
+                            .padding(8.dp)
+                    ) {
+                        StreetViewSwitch(title = "Panning", checked = isPanningEnabled) {
+                            isPanningEnabled = it
+                        }
+                        StreetViewSwitch(title = "Zooming", checked = isZoomEnabled) {
+                            isZoomEnabled = it
+                        }
                     }
-                    StreetViewSwitch(title = "Zooming", checked = isZoomEnabled) {
-                        isZoomEnabled = it
-                    }
+                } else {
+                    Text("Location not available.")
                 }
             }
         }

--- a/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
+++ b/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
@@ -81,6 +81,7 @@ class StreetViewActivity : ComponentActivity() {
                         }
                 }
                 launch {
+                    // Make sure that this API Key has Street View Static API activated.
                     streetViewResult =
                         fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
                 }

--- a/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
+++ b/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
@@ -40,21 +40,32 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.unit.dp
 import com.google.android.gms.maps.StreetViewPanoramaOptions
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.Status
 import com.google.maps.android.compose.streetview.StreetView
 import com.google.maps.android.compose.streetview.rememberStreetViewCameraPositionState
 import com.google.maps.android.ktx.MapsExperimentalFeature
 import kotlinx.coroutines.launch
+import com.google.maps.android.StreetViewUtils.Companion.fetchStreetViewData
 
 class StreetViewActivity : ComponentActivity() {
 
     private val TAG = StreetViewActivity::class.java.simpleName
 
+    // This is an invalid location. If you use it instead of Singapore, the StreetViewUtils
+    // will return NOT_FOUND.
+    val invalidLocation = LatLng(32.429634, -96.828891)
+
     @OptIn(MapsExperimentalFeature::class)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+
         setContent {
             var isPanningEnabled by remember { mutableStateOf(false) }
             var isZoomEnabled by remember { mutableStateOf(false) }
+            var streetViewResult by remember { mutableStateOf(Status.NOT_FOUND) }
+
             val camera = rememberStreetViewCameraPositionState()
             LaunchedEffect(camera) {
                 launch {
@@ -69,23 +80,29 @@ class StreetViewActivity : ComponentActivity() {
                             Log.d(TAG, "Location at: $it")
                         }
                 }
+                launch {
+                    streetViewResult =
+                        fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
+                }
             }
             Box(Modifier.fillMaxSize(), Alignment.BottomStart) {
-                StreetView(
-                    Modifier.matchParentSize(),
-                    cameraPositionState = camera,
-                    streetViewPanoramaOptionsFactory = {
-                        StreetViewPanoramaOptions().position(singapore)
-                    },
-                    isPanningGesturesEnabled = isPanningEnabled,
-                    isZoomGesturesEnabled = isZoomEnabled,
-                    onClick = {
-                        Log.d(TAG, "Street view clicked")
-                    },
-                    onLongClick = {
-                        Log.d(TAG, "Street view long clicked")
-                    }
-                )
+                if (streetViewResult == Status.OK) {
+                    StreetView(
+                        Modifier.matchParentSize(),
+                        cameraPositionState = camera,
+                        streetViewPanoramaOptionsFactory = {
+                            StreetViewPanoramaOptions().position(singapore)
+                        },
+                        isPanningGesturesEnabled = isPanningEnabled,
+                        isZoomGesturesEnabled = isZoomEnabled,
+                        onClick = {
+                            Log.d(TAG, "Street view clicked")
+                        },
+                        onLongClick = {
+                            Log.d(TAG, "Street view long clicked")
+                        }
+                    )
+                }
                 Column(
                     Modifier
                         .fillMaxWidth()
@@ -103,6 +120,7 @@ class StreetViewActivity : ComponentActivity() {
         }
     }
 }
+
 
 @Composable
 fun StreetViewSwitch(title: String, checked: Boolean, onCheckedChange: (Boolean) -> Unit) {

--- a/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
+++ b/app/src/main/java/com/google/maps/android/compose/StreetViewActivity.kt
@@ -80,7 +80,8 @@ class StreetViewActivity : ComponentActivity() {
                         }
                 }
                 launch {
-                    // Make sure that this API Key has Street View Static API activated.
+                    // Be sure to enable the Street View Static API on the project associated with
+                    // this API key using the instructions at https://goo.gle/enable-sv-static-api
                     streetViewResult =
                         fetchStreetViewData(singapore, BuildConfig.MAPS_API_KEY)
                 }

--- a/maps-compose/src/main/java/com/google/maps/android/compose/streetview/StreetView.kt
+++ b/maps-compose/src/main/java/com/google/maps/android/compose/streetview/StreetView.kt
@@ -1,3 +1,17 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package com.google.maps.android.compose.streetview
 
 import android.content.ComponentCallbacks
@@ -23,14 +37,17 @@ import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleEventObserver
 import com.google.android.gms.maps.StreetViewPanoramaOptions
 import com.google.android.gms.maps.StreetViewPanoramaView
-import com.google.android.gms.maps.model.StreetViewPanoramaCamera
 import com.google.android.gms.maps.model.StreetViewPanoramaOrientation
 import com.google.maps.android.compose.disposingComposition
 import com.google.maps.android.ktx.MapsExperimentalFeature
 import com.google.maps.android.ktx.awaitStreetViewPanorama
 
 /**
- * A composable for displaying a Street View for a given location.
+ * A composable for displaying a Street View for a given location. A location might not be available for a given
+ * set of coordinates. We recommend you to check our sample on [StreetViewActivity] using our utility function
+ * in [StreetViewUtils] to manage non-existing locations.
+ *
+ *
  *
  * @param modifier Modifier to be applied to the StreetView
  * @param cameraPositionState the [StreetViewCameraPositionState] to be used to control or observe

